rust-symcrypt/src/rsa/pkcs1.rs

```diff
@@ -0,0 +1,348 @@
+//! use symcrypt::rsa::{RsaKeyPair, RsaKeyUsage, RsaPublicKey};
+//! use symcrypt::hash::{sha256, HashAlgorithm};
+//!
+//! // Generate key pair.
+//! let key_pair = RsaKeyPair::generate_new(2048, None, RsaKeyUsage::SignAndEncrypt).unwrap();
+//!
+//! // Set up message.
+//! let hashed_message = sha256(b"hello world");
+//! let hash_algorithm = HashAlgorithm::Sha256;
+//!
+//! // Create signature.
+//! let signature = key_pair.pkcs1_sign(&hashed_message, hash_algorithm).unwrap();
+//!
+//! // Create Public Key to verify signature.
+//! let public_key_blob = key_pair.export_public_key_blob().unwrap();
+//! let public_key = RsaPublicKey::set_public_key(&public_key_blob.modulus, &public_key_blob.pub_exp, RsaKeyUsage::SignAndEncrypt).unwrap();
+//!
+//! // Verify signature.
+//! let verify_result = public_key.pkcs1_verify(&hashed_message, &signature, hash_algorithm);
+//! assert!(verify_result.is_ok());
+//! ```
+//!
+//! ## Encrypt and Decrypt using RsaKeyPair
+//!
+//! ```rust
+//! use symcrypt::rsa::{RsaKeyPair, RsaKeyUsage};
+//!
+//! // Generate key pair.
+//! let key_pair = RsaKeyPair::generate_new(2048, None, RsaKeyUsage::SignAndEncrypt).unwrap();
+//!
+//! // Set up message.
+//! let message = b"example message";
+//!
+//! // Encrypt message.
+//! let encrypted_message = key_pair.pkcs1_encrypt(message).unwrap();
+//!
+//! // Decrypt message.
+//! let decrypted_message = key_pair.pkcs1_decrypt(&encrypted_message).unwrap();
+//! assert_eq!(decrypted_message, message);
+//! ```
+//!
+use crate::errors::SymCryptError;
+use crate::hash::HashAlgorithm;
+use crate::rsa::{RsaKeyPair, RsaPublicKey};
+use crate::NumberFormat;
+
+/// Impl for Pkcs1 RSA via [`RsaKeyPair`]
+impl RsaKeyPair {
+    /// `pcks1_sign()` is only available for [`RsaKeyPair`].
+    ///
+    /// This function signs a hashed message using the private key of the key pair and returns a `Vec<u8>` representing the signature,
+    /// or a [`SymCryptError`] if the operation fails.
+    ///
+    /// `hashed_message` is a `&[u8]` representing the hashed message to be signed.
+    ///
+    /// `hash_algorithm` is a [`HashAlgorithm`] representing the hash algorithm used to hash the message.
+    pub fn pkcs1_sign(
+        &self,
+        hashed_message: &[u8],
+        hash_algorithm: HashAlgorithm,
+    ) -> Result<Vec<u8>, SymCryptError> {
+        let mut result_size = 0;
+        let modulus_size = self.get_size_of_modulus();
+        let mut signature = vec![0u8; modulus_size as usize];
+        let converted_hash_oids = hash_algorithm.to_oid_list();
+        unsafe {
+            // SAFETY: FFI calls
+            let error_code = symcrypt_sys::SymCryptRsaPkcs1Sign(
+                self.inner(),
+                hashed_message.as_ptr(),
+                hashed_message.len() as symcrypt_sys::SIZE_T,
+                converted_hash_oids.as_ptr(),
+                converted_hash_oids.len() as symcrypt_sys::SIZE_T,
+                symcrypt_sys::SYMCRYPT_FLAG_RSA_PKCS1_NO_ASN1,
```

Comment:
> symcrypt_sys::SYMCRYPT_FLAG_RSA_PKCS1_NO_ASN1, [](http://example.com/codeflow?start=15&length=47)

You don't want to set. You are setting the ASN.1 OID in the previous parameters.